aggregator/src/aggregator/aggregation_job_driver.rs

```diff
@@ -609,68 +609,37 @@ where
-                            if let Err(err) = accumulator.update(
-                                aggregation_job.partial_batch_identifier(),
-                                stepped_aggregation.report_aggregation.report_id(),
-                                stepped_aggregation.report_aggregation.time(),
-                                &output_share,
-                            ) {
-                                warn!(
-                                    report_id = %stepped_aggregation.report_aggregation.report_id(),
-                                    ?err,
-                                    "Could not update batch aggregation",
-                                );
-                                self.aggregate_step_failure_counter
-                                    .add(1, &[KeyValue::new("type", "accumulate_failure")]);
-                                ReportAggregationState::<SEED_SIZE, A>::Failed {
-                                    prepare_error: PrepareError::VdafPrepError,
-                                }
-                            } else {
-                                ReportAggregationState::Finished
-                            }
+                            (ReportAggregationState::Finished, Some(output_share))
+                        }
+                        Err(prepare_error) => {
+                            (ReportAggregationState::Failed { prepare_error }, None)
                         }
-                        Err(prepare_error) => ReportAggregationState::Failed { prepare_error },
                     }
                 }
 
                 PrepareStepResult::Finished => {
-                    if let PingPongState::Finished(output_share) = &stepped_aggregation.leader_state
+                    if let PingPongState::Finished(output_share) = stepped_aggregation.leader_state
                     {
                         // Helper finished and we had already finished. Commit the output share.
-                        if let Err(err) = accumulator.update(
-                            aggregation_job.partial_batch_identifier(),
-                            stepped_aggregation.report_aggregation.report_id(),
-                            stepped_aggregation.report_aggregation.time(),
-                            output_share,
-                        ) {
-                            warn!(
-                                report_id = %stepped_aggregation.report_aggregation.report_id(),
-                                ?err,
-                                "Could not update batch aggregation",
-                            );
-                            self.aggregate_step_failure_counter
-                                .add(1, &[KeyValue::new("type", "accumulate_failure")]);
-                            ReportAggregationState::<SEED_SIZE, A>::Failed {
-                                prepare_error: PrepareError::VdafPrepError,
-                            }
```

Comment:
I understand that this failure path goes away because we don't use `Accumulator` anymore. But we also are no longer incrementing metrics with the `accumulate_failure` label. Should we plumb that into `AggregationJobWriter`?

Reply:
Good point -- I stared at this for a while and things would have to go very wrong for this merge to fail, but that means that if something does go wrong we'll probably appreciate having the metric.